virtool/custom_oas/oas/view.py

```diff
@@ -0,0 +1,245 @@
+import typing
+import copy
+from inspect import getdoc
+from itertools import count
+from typing import List, Type, Optional, get_type_hints
+
+from aiohttp.web import Response, json_response
+from aiohttp.web_app import Application
+from pydantic import BaseModel
+
+from aiohttp_pydantic.oas.struct import OpenApiSpec3, OperationObject, PathItem
+
+from aiohttp_pydantic.oas import docstring_parser
+
+from aiohttp_pydantic.injectors import _parse_func_signature
+
+from aiohttp_pydantic.utils import is_pydantic_base_model
+
+from aiohttp_pydantic.view import PydanticView, is_pydantic_view
+
+from aiohttp_pydantic.oas.typing import is_status_code_type
+
+
+class _OASResponseBuilder:
+    """
+    Parse the type annotated as returned by a function and
+    generate the OAS operation response.
+    """
+
+    def __init__(self, oas: OpenApiSpec3, oas_operation, status_code_descriptions):
+        self._oas_operation = oas_operation
+        self._oas = oas
+        self._status_code_descriptions = status_code_descriptions
+
+    def _handle_pydantic_base_model(self, obj):
+        if is_pydantic_base_model(obj):
+            response_schema = obj.schema(
+                ref_template="#/components/schemas/{model}"
+            ).copy()
+            if def_sub_schemas := response_schema.pop("definitions", None):
+                self._oas.components.schemas.update(def_sub_schemas)
+            return response_schema
+        return {}
+
+    def _handle_list(self, obj):
+        if typing.get_origin(obj) is list:
+            return {
+                "type": "array",
+                "items": self._handle_pydantic_base_model(typing.get_args(obj)[0]),
+            }
+        return self._handle_pydantic_base_model(obj)
+
+    def _handle_status_code_type(self, obj, *args):
```

Comment:
Main changes are lines 53-75. Pass in *args which is the request body, such as {"group_id": "research"}. Then modify
the example response to reflect the request.